HM3CoreKill/HM3CoreKill/ck/HM3InGameTools.cpp

```diff
@@ -330,6 +332,55 @@ namespace ck
 					ImGui::Text("Location pointer at 0x%.8X", currentActor->ActorInformation->location);
 					ImGui::Text("Position:"); ImGui::SameLine(0.f, 4.f); ImGui::InputFloat3("", actorPosition); 
 
+					{
+						ImGui::Text("Set animation: "); ImGui::SameLine(0.f, 5.f);
+
+						static std::string item_current = "(None)";            // Here our selection is a single pointer stored outside the object.
+						static ioi::hm3::ZAnimationInfo* current_anim = nullptr;
+
+						if (ImGui::BeginCombo("CXX", item_current.c_str())) // The second parameter is the label previewed before opening the combo.
+						{
+							const ck::HM3AnimationRegistry& registry = ck::HM3AnimationRegistry::getRegistry();
+							std::vector<ioi::hm3::ZAnimationInfo*> animations;
+							registry.getLoadedAnimations(animations);
+
+							for (int i = 0; i < animations.size(); i++)
+							{
+								if (!animations[i])
+									continue;
+
+								bool is_selected = (item_current == animations[i]->m_name);
+								if (ImGui::Selectable(animations[i]->m_name, is_selected))
+								{
+									item_current = animations[i]->m_name;
+									current_anim = animations[i];
+								}
+
+								if (is_selected)
+									ImGui::SetItemDefaultFocus();
+							}
+
+							ImGui::EndCombo();
+						}
+						ImGui::SameLine(0.f, 5.f);
+						if (current_anim && ImGui::Button("Apply"))
+						{
+							DWORD pFunc0 = HM3Function::getVirtualFunctionAddress((DWORD)currentActor, 0x164);
```

Comment:
This is a part of ZHM3Actor API. We must use methods from vftable directly through the class. Need to declare that functions and use it without "calculated calls".